node/get_version_info/action.yml

```diff
@@ -0,0 +1,19 @@
+
+name: Publish Release Asset to S3
+description: "Publish Asset to S3"
+inputs:
+  npm_package_name:
+    description: the npm package name
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Get release version and release package file name
+      id: get_version
+      shell: bash
+      run: |
+        package_version=$(jq --raw-output '.version' package.json)
+        echo "package_version=${package_version}" >> "$GITHUB_ENV"
```

Comment:
ENV is easier to work with because anything can be accessed with `env.<NAME>` and doesn't need to worry about step ids.

Reply:
It may be easier to use but it removes the explicit linkages of dependence between steps, the need to reference ids is a feature not a burden

Reply:
Note that as per the [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions), steps don't have outputs. The syntax `steps.<step-id>.outputs.<output-id>` is used to access outputs from an action called in that particular step. The `$GITHUB_OUTPUT` env variable is shared between all steps in a job, so it's no different from `$GITHUB_ENV`, with the subtle difference that `$GITHUB_ENV` is not made available as a job output.

Reply:
> steps don't have outputs.

Sorry a bit confused here, each step can have outputs, the [`steps` context](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context) is how you access them. [Here's an example of how we use them in the driver](https://github.com/mongodb/node-mongodb-native/blob/main/.github/workflows/release_notes.yml#L66)

> used to access outputs from an action called in that particular step

When you've called away to an action, it can return specified values by declaring its outputs: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs-for-composite-actions 

Whether you're calling away to an external action or running an inline script step using outputs is an alternative to using `env`. The benefit of outputs is the requirement to declaratively connect your ins and outs.

Reply:
> [...] the [`steps` context](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context) is how you access them

Welp, of course it'd be in the step context, not in the workflow syntax. I stand corrected - I assumed the outputs were only available when calling to a separate action in a step.